MAVProxy/modules/mavproxy_park.py

```diff
@@ -0,0 +1,113 @@
+        self.parked = False
+        self.radius = 2  # max distance
+        self.check_interval = 1  # 1=check every second
+        self.last_check = time.time()
+        self.notify_interval = 4  # minimum seconds between notifications
+        self.last_notification = time.time()
+        self.dist = 0
+        self.usage = "Usage: park <status|on|off|radius>\n On: enable departure warning.\n Radius defines the" \
+                     " distance in meters (3D) from parked position, when exceeded an alarm is raised."
+#        self.park_settings = mp_settings.MPSettings(
+#            [('verbose', bool, False),('versee', bool, True),])
+        self.add_command('park', self.cmd_park, "park module")
+
+    def cmd_park(self, args):
+        """control behaviour of the module"""
+        if len(args) == 0:
+            print(self.usage)
+        elif args[0] == "status":
+            self.status()
+        elif args[0] == "on":
+            self.park_on()
+        elif args[0] == "off":
+            self.park_off()
+        elif args[0] == "radius":
+            if len(args) < 2:
+                print("Usage: park radius <RADIUS>")
+                return
+            self.radius = float(args[1])
+        else:
+            print(self.usage)
+
+    def status(self):
+        """returns information about module"""
+        if self.parked:
+            print("Parked at lat:", self.lat, "lon:", self.lon, "alt:", self.alt, "max radius", self.radius, "m")
+        else:
+            print("Not parked")
+            
+    def park_on(self):
+        """set park"""
+        self.latp = self.lat
+        self.lonp = self.lon
+        self.altp = self.alt
+        self.parked = True
+        print("Parked at lat:", self.lat, "lon:", self.lon, "alt:", self.alt,
+              "you will be notified if it moves >", self.radius, "m")
+
+    def park_off(self):
+        self.parked = False
+        print("Park warning is off")
+
+    def idle_task(self):
+        """called frequently by mavproxy"""
+        now = time.time()
+        if self.parked and now-self.last_check > self.check_interval:
+            self.last_check = now
+            self.dist = self.haversine(self.lon, self.lat, self.alt, self.lonp, self.latp, self.altp)
+            if self.dist > self.radius and now-self.last_notification > self.notify_interval:
+                self.last_notification = now
+                message = "Vehicle moved "
+                message2 = message + "%.1f" % self.dist + "meters"
+                self.say("%s: %s" % (self.name, message2))
+                self.master.mav.statustext_send(mavutil.mavlink.MAV_SEVERITY_NOTICE, message2)
+
+    def mavlink_packet(self, m):
+        """handle mavlink packets"""
+        if m.get_type() == 'GLOBAL_POSITION_INT':
+            self.lat = m.lat * 1.0e-7
+            self.lon = m.lon * 1.0e-7
+            self.alt = m.relative_alt * 1.0e-3
+
+    def haversine(self, lon1, lat1, alt1, lon2, lat2, alt2):
```

Comment:
you can just call mp_util.gps_distance() to get the horizontal component

Reply:
@tridge now I remember:  I did not use mp_util.gps_distance() because it only calculated lateral distance, while I wished to have a 3D sphere-shaped shape - for correct warning in all directions.
Of course, I could check for altitude separately, ending with a cylindrical threshold (tripping on a lateral distance or vertical deviation.) 
Are there drawbacks of doing it the way I did?

Reply:
it looks like you calculate horizontal then add vertical, so you can use gps_distance for horizontal and add vertical in the same way
I just like all our earth model fns to be consistent, I'm not keen on having two different methods in the code for this distance.

Reply:
Trank you for replying, I'll change it.

Reply:
@tridge   - Please accept the fixed version.